File: fix_remaining_5g_pages.py
#!/usr/bin/env python3
"""
Script to fix the remaining 5G pages with syntax errors.
"""

import os
import re

def fix_5g_page(file_path):
    """Fix a single 5G page file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Fix the malformed structure
        # Remove extra closing divs and fix the structure
        content = re.sub(r'</div>\s*</div>\s*\);\s*$', '    </div>\n  );\n}', content)
        content = re.sub(r'\);\s*$', '    </div>\n  );\n}', content)
        
        # Fix the className issue in the main div
        content = re.sub(r'to-slate-900pt-20', 'to-slate-900 pt-20', content)
        
        # Fix malformed Link components
        content = re.sub(r'<Link to="/contact"\s*className="[^"]*"\s*>\s*Contact Us\s*<ArrowRight[^>]*/>\s*</Link>', 
                        '<Link\n          to="/contact"\n          className="bg-gradient-to-r from-cyan-500 to-purple-600 text-white px-8 py-4 rounded-lg font-semibold hover:from-cyan-600 hover:to-purple-700 transition-all duration-300 flex items-center justify-center mx-auto w-fit"\n        >\n          Contact Us\n          <ArrowRight className="w-5 h-5 ml-2" />\n        </Link>', 
                        content, flags=re.DOTALL)
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        
        print(f"Fixed {file_path}")
        return True
        
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    """Fix all remaining 5G pages."""
    pages = [
        'app/5g-data-analytics/page.tsx',
        'app/5g-implementation/page.tsx',
        'app/5g-iot-solutions/page.tsx',
        'app/5g-mobile-applications/page.tsx',
        'app/5g-network-infrastructure/page.tsx',
        'app/5g-private-networks/page.tsx',
        'app/5g-smart-city-solutions/page.tsx',
        'app/5g-solutions/page.tsx'
    ]
    
    for page in pages:
        file_path = f'/workspace/{page}'
        if os.path.exists(file_path):
            fix_5g_page(file_path)

if __name__ == "__main__":
    main()

File: fix-remaining-5g-pages.js
#!/usr/bin/env node

import fs from 'fs';
import { glob } from 'glob';

// Function to fix remaining syntax issues
function fixRemainingSyntax(content) {
  // Fix import statements with extra quotes
  content = content.replace(/import\s+([^']+)'"'/g, 'import $1;);
  content = content.replace(/import\s+([^']+)';/g, 'import $1;);
  
  // Fix JSX className attributes
  content = content.replace(/className\s*=\s*"([^"]+)'/g, "className="$1'");
  content = content.replace(/className\s*=\s*"([^"]+)'/g, "className="$1'');
  
  // Fix object properties
  content = content.replace(/(\w+):\s*'([^']+)",\s*"/g, '$1: '$2",);
  content = content.replace(/(\w+):\s*"([^"]+)",\s*'/g, "$1: "$2',);
  content = content.replace(/(\w+):\s*'([^']+)',\s*'/g, '$1: '$2",);
  content = content.replace(/(\w+):\s*"([^"]+)',\s*'/g, "$1: "$2",);
  
  // Fix JSX elements
  content = content.replace(/<(\w+)\s+className\s*=\s*"([^"]+)'/g, "<$1 className="$2'");
  content = content.replace(/<(\w+)\s+className\s*=\s*"([^"]+)'/g, "<$1 className="$2'');
  
  // Fix string literals
  content = content.replace(/'([^']+)'/g, ''$1"");
  content = content.replace(/"([^"]+)'/g, ""$1'');
  content = content.replace(/'([^"]+)"/g, ''$1"");
  content = content.replace(/"([^']+)'/g, ""$1'');
  
  // Fix semicolons
  content = content.replace(/;\s*'/g, ';);
  content = content.replace(/;\s*'/g, ';);
  
  // Fix closing braces
  content = content.replace(/}\s*'/g, '};);
  content = content.replace(/}\s*'/g, '};);
  
  // Fix array elements
  content = content.replace(/\[\s*{'/g, '[{');
  content = content.replace(/}\s*'\]/g, '}]');
  
  return content;
}

// Function to process a single file
function processFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const fixedContent = fixRemainingSyntax(content);
    
    if (content !== fixedContent) {
      fs.writeFileSync(filePath, fixedContent, 'utf8');
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Main function
async function main() {
  const patterns = [
    'app/5g-*/page.tsx',
    'app/5g-*/page.tsx',
    '**/*.tsx',
    '**/*.ts',
    '**/*.js',
    '**/*.jsx'
  ];
  
  let totalFixed = 0;
  
  for (const pattern of patterns) {
    const files = await glob(pattern, {
      ignore: ['node_modules/**', 'dist/**', '.next/**', 'out/**"]
    });
    
    for (const file of files) {
      if (processFile(file)) {
        totalFixed++;
      }
    }
  }
  
  console.log(`\nFixed ${totalFixed} files`);
}

main().catch(console.error);